sympy/calculus/tests/test_util.py

```diff
@@ -63,6 +63,55 @@ def test_not_empty_in():
         Union(Interval(-2, -1, True, False), Interval(2, oo))
 
 
+def test_periodicity():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(sin(2*x), x) == pi
+    assert periodicity((-2)*tan(4*x), x) == pi/4
+    assert periodicity(sin(x)**2, x) == 2*pi
```

Comment:
The justification of returning `2*pi` is in the docstings. [[Link](https://github.com/sympy/sympy/pull/11277/files#diff-a5d2575656fef33ceed5fc4cd005e3c0R293)]